plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergV2.java

```diff
@@ -175,12 +180,69 @@ public void testV2TableWithEqualityDelete()
+        writeEqualityDeleteToNationTable(icebergTable, Optional.of(icebergTable.spec()));
+        List<String> initialActiveFiles = getActiveFiles(tableName);
+        query("ALTER TABLE " + tableName + " EXECUTE OPTIMIZE");
+        assertQuery("SELECT * FROM " + tableName, "SELECT * FROM nation WHERE regionkey != 1");
+        // natiokey is before the equality delete column in the table schema, comment is after
+        assertQuery("SELECT nationkey, comment FROM " + tableName, "SELECT nationkey, comment FROM nation WHERE regionkey != 1");
+        Assertions.assertThat(loadTable(tableName).currentSnapshot().summary().get("total-equality-deletes")).isEqualTo("0");
+        List<String> updatedFiles = getActiveFiles(tableName);
+        Assertions.assertThat(updatedFiles).doesNotContain(initialActiveFiles.toArray(new String[0]));
+    }
+
+    @Test
+    public void testOptimizingV2TableDoesntRemoveEqualityDeletesWhenOnlyOnePartitionIsOptimized()
+            throws Exception
+    {
+        String tableName = "test_v2_equality_delete" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " WITH (partitioning = ARRAY['regionkey']) AS SELECT * FROM tpch.tiny.nation", 25);
+        Table icebergTable = updateTableToV2(tableName);
+        Assertions.assertThat(icebergTable.currentSnapshot().summary().get("total-equality-deletes")).isEqualTo("0");
+        List<String> initialActiveFiles = getActiveFiles(tableName);
+        writeEqualityDeleteToNationTable(icebergTable, Optional.of(icebergTable.spec()));
+        query("ALTER TABLE " + tableName + " EXECUTE OPTIMIZE WHERE regionkey != 1");
+        assertQuery("SELECT * FROM " + tableName, "SELECT * FROM nation WHERE regionkey != 1");
+        // natiokey is before the equality delete column in the table schema, comment is after
+        assertQuery("SELECT nationkey, comment FROM " + tableName, "SELECT nationkey, comment FROM nation WHERE regionkey != 1");
+        Assertions.assertThat(loadTable(tableName).currentSnapshot().summary().get("total-equality-deletes")).isEqualTo("1");
+        List<String> updatedFiles = getActiveFiles(tableName);
+        Assertions.assertThat(updatedFiles).doesNotContain(initialActiveFiles.stream().filter(path -> !path.contains("regionkey=1")).toArray(String[]::new));
+    }
+
+    @Test
+    public void testOptimizingV2TableWithEmptyPartitionSpec()
+            throws Exception
+    {
+        String tableName = "test_v2_equality_delete" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT * FROM tpch.tiny.nation", 25);
+        Table icebergTable = updateTableToV2(tableName);
+        Assertions.assertThat(icebergTable.currentSnapshot().summary().get("total-equality-deletes")).isEqualTo("0");
+        writeEqualityDeleteToNationTable(icebergTable, Optional.empty());
+        List<String> initialActiveFiles = getActiveFiles(tableName);
+        query("ALTER TABLE " + tableName + " EXECUTE OPTIMIZE");
```

Comment:
Can you add more more check here where you optimize with a predicate and show the equality delete is not removed?

Reply:
Sorry I am not sure I get this. I have different method `testOptimizingV2TableDoesntRemoveEqualityDeletesWhenOnlyPartOfTheTableIsOptimized` that checks optimization with predicate where I validate that equality delete is preserved

Reply:
That one is an equality delete with a partition but you're optimizing different partitions. You don't have one with a global equality delete where only part of the table is optimized

Reply:
Ok so I am not able to do anything better than this:
```
@Test
    public void testOptimizingPartitionsOfV2TableWithGlobalEqualityDeleteFile()
            throws Exception
    {
        String tableName = "test_v2_equality_delete" + randomTableSuffix();
        assertUpdate("CREATE TABLE " + tableName + " WITH (partitioning = ARRAY['regionkey']) AS SELECT * FROM tpch.tiny.nation", 25);
        Table icebergTable = updateTableToV2(tableName);
        Assertions.assertThat(icebergTable.currentSnapshot().summary().get("total-equality-deletes")).isEqualTo("0");
        writeEqualityDeleteToNationTable(icebergTable, Optional.of(icebergTable.spec()), Optional.of(new PartitionData(new Long[]{1L, 2L, 3L, 4L})));
        List<String> initialActiveFiles = getActiveFiles(tableName);
        assertQuery("SELECT * FROM " + tableName, "SELECT * FROM nation WHERE regionkey != 1");
        query("ALTER TABLE " + tableName + " EXECUTE OPTIMIZE WHERE regionkey != 1");
        assertQuery("SELECT * FROM " + tableName, "SELECT * FROM nation WHERE regionkey != 1");
        // natiokey is before the equality delete column in the table schema, comment is after
        assertQuery("SELECT nationkey, comment FROM " + tableName, "SELECT nationkey, comment FROM nation WHERE regionkey != 1");
        Assertions.assertThat(loadTable(tableName).currentSnapshot().summary().get("total-equality-deletes")).isEqualTo("1");
        List<String> updatedFiles = getActiveFiles(tableName);
        Assertions.assertThat(updatedFiles)
                .doesNotContain(initialActiveFiles.stream()
                                .filter(path -> !path.contains("regionkey=1"))
                                .collect(toImmutableList())
                                .toArray(new String[0]));
    }
```

maybe I am doing something wrong but creating DeleteFiles with empty spec and empty partition for partitioned table(so without 
```
.withSpec(partitionSpec.get())
 .withPartition(partitionData.get()); for partitioned table makes trino ignore these delete files . 
```

) makes trino ignore that delete file

Reply:
There is really nothing to figure out, or I am missing something.
When you have partitioned table you have to provide partition data(`withSpec` and `withPartition`) to delete files  otherwise you can't create delete files due to `java.lang.IllegalArgumentException: Partition must not be null for partitioned writes` . 
I must have made mixed something up while writing previous comment.